errdefs/helpers_test.go

```diff
@@ -45,6 +53,9 @@ func TestForbidden(t *testing.T) {
 	if !IsForbidden(e) {
 		t.Fatalf("expected forbidden error, got: %T", e)
 	}
+	if !IsKnownErrorType(e) {
```

Comment:
This looks redundant; we checked if it's a `Forbidden` error above, and otherwise the test fails (in which case it prints the `%T`)

Reply:
right, but this test is to make sure that `Forbidden` is a known error type.

Reply:
Not sure I understand; `IsForbidden` checks that it's a known error type (must be `ErrForbidden`); https://github.com/moby/moby/blob/8e610b2b55bfd1bfa9436ab110d311f5e8a74dcb/errdefs/is.go#L62-L66

So when we reach this line, we know it's a `ErrForbidden`, thus a known error-type?

Reply:
the test isn't for `IsForbidden`, it's for `IsKnownErrorType`.

Reply:
If it's to test if `IsKnownErrorType` is handling all error-types, it should be a separate test. This test is to test if `IsForbidden` is working as expected.

But I'm still not sure what the `IsKnownErrorType` utility is solving (see my comment above: https://github.com/moby/moby/pull/38770/files#r263735447)

By wrapping "unknown error types" as an `errdefs.Unknown`, we're actually _masking_ unhandled errors, and won't be logging them as errors that have not been mapped;

https://github.com/moby/moby/blob/72b0b0387d777328882b02224bb78ad59c0837e3/api/server/httputils/errors.go#L50-L66

Reply:
Yeah, I see that now. I'm gonna change it when I spend some time working on this PR.